app/gosns/gosns.go

```diff
@@ -485,6 +491,132 @@ func DeleteTopic(w http.ResponseWriter, req *http.Request) {
 
 }
 
+func PublishBatch(w http.ResponseWriter, req *http.Request) {
+	content := req.FormValue("ContentType")
+	topicArn := req.FormValue("TopicArn")
+
+	arnSegments := strings.Split(topicArn, ":")
+	topicName := arnSegments[len(arnSegments)-1]
+	topic, ok := app.SyncTopics.Topics[topicName]
+	if !ok {
+		createErrorResponse(w, req, "TopicNotFound")
+		return
+	}
+
+	batchMessageIdToMessageBody := make(map[string]string, 10)
+	batchMessageIdToMessageStructure := make(map[string]string, 10)
+	batchMessageIdToMessageAttributes := make(map[string]map[string]app.MessageAttributeValue, 10)
+	batchMessageIdToSubject := make(map[string]string, 10)
+
+	for memberIndex := 1; true; memberIndex++ {
+		if memberIndex > 10 {
```

Comment:
So we have an implied limit of 10 on this?  Would it be worth doing a couple things for user clarity?
1. Return that in the errors?  `TooMany` is good, but would something like `Over10EntriesRequested` be more clear?  (Also, I'm not married to the camel case errors, especially in the responses - I know that's the current pattern  in here, but I leave you the option if you prefer.)
2. Is there a way we could bump this `Max 10` validation up before the loops and return it there?  If so, would that make it clearer when we loop by doing something like 
     `memberIndex := 1; memberIndex < len(members); memberIndex++`
If that works would that make it more clear what's happening in these loops?

Reply:
It's an actual SNS limit - see [here](https://docs.aws.amazon.com/sns/latest/api/API_PublishBatch.html)
My approach has been to largely mimic the implemented SNS:Publish (gosns.Publish) and SQS:SendMessageBatch (gosqs...) for consistency.

Reply:
Ah nice!  I see your entry in the error message now, so I think that serves.  Thanks!